### PR TITLE
Fix client-helpers `cleanId` function to properly clean the entire string

### PR DESCRIPTION
**Pull request by TheWisestOne**
The current function will only replace the first occurence of an invalid character. This fixes it to handle multiple invalid characters. Also, update the unit test for `client.ts` to exercise this code.

This resolves https://github.com/RobinTail/express-zod-api/issues/805

### CODE REVIEW

File: src/client-helpers.ts [hunk 1]
```diff
@@ -200,7 +200,8 @@ export const cleanId = (path: string, method: string, suffix: string) => {
   return [method]
     .concat(path.split("/"))
     .concat(suffix)
-    .map((entry) => entry.replace(/[^A-Z0-9]/i, ""))
+    .map((entry) => entry.split(/[^A-Z0-9]/gi))
```
Comment (TheWisestOne) [inline src/client-helpers.ts:203]: the addition of the `g` allows the clean function to affect all invalid characters
The use of `.split` followed by `.flat` allows for the `CapitalCasing` that happens next to correctly map these `special-character_separated` words into capital case correctly: `SpecialCharacterSeparated`

File: src/client-helpers.ts [hunk 2]
```diff
@@ -200,7 +200,8 @@ export const cleanId = (path: string, method: string, suffix: string) => {
...
+    .flat(1)
```
Comment (RobinTail) [inline src/client-helpers.ts:204]:
```suggestion
    .flat()
```

`1` [is the default](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat#parameters)

File: src/client-helpers.ts (hunk 1 above)
Comment (RobinTail) [reply src/client-helpers.ts:203]: ok

Comment (RobinTail) [reply src/client-helpers.ts:203]: there is also `flatMap`, which is described are more efficient

> It is identical to a map() followed by a flat() of depth 1 , but slightly more efficient than calling those two methods separately.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap

File: src/client-helpers.ts [hunk 3]
```diff
@@ -200,7 +200,8 @@ export const cleanId = (path: string, method: string, suffix: string) => {
   return [method]
     .concat(path.split("/"))
     .concat(suffix)
-    .map((entry) => entry.replace(/[^A-Z0-9]/i, ""))
+    .map((entry) => entry.split(/[^A-Z0-9]/gi))
+    .flat()
```
Comment (RobinTail) [inline src/client-helpers.ts:204]:
```suggestion
    .flatMap((entry) => entry.split(/[^A-Z0-9]/gi))
```

there is also `flatMap`, which is described as more efficient

> It is identical to a map() followed by a flat() of depth 1 , but slightly more efficient than calling those two methods separately.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap

**Review by RobinTail — APPROVED** [review]
Great!

**Pull request merged**